### PR TITLE
`prefer-number-properties`: Don't require by default for `Infinity`/`-Infinity` to be written as `Number.POSITIVE_INFINITY`/`Number.NEGATIVE_INFINITY`

### PR DESCRIPTION
**Pull request by Richienb**
I don't think it's a breaking change

Related to #2311

### CODE REVIEW

File: docs/rules/prefer-number-properties.md [hunk 1]
```diff
@@ -89,9 +89,9 @@ Type: `object`
 ### checkInfinity
 
 Type: `boolean`\
```
Comment (sindresorhus) [inline docs/rules/prefer-number-properties.md:91]: You also need to update line 82 and 47

**Comment by sindresorhus** [comment]
> I don't think it's a breaking change

Anything that changes the lint results is a breaking change.

**Comment by sindresorhus** [comment]
CI is failing

**Comment by Richienb** [comment]
Not sure what to do about the CI

**Comment by sindresorhus** [comment]
May be similar to https://github.com/sindresorhus/eslint-plugin-unicorn/pull/2186#issuecomment-1912370906

**Review by sindresorhus — APPROVED** [review]

**Review by fisker — APPROVED** [review]
I'm fine with the change, since the option was kept.

**Pull request merged**